### DISCUSSION TITLE
プッシュ通知でFCMデバイス登録トークンを更新する理由について

### DISCUSSION THREAD

**Discussion by ghost** [ws-4020/mobile-app-crib-notes#739] [Q&A] [answered]
④の再生成の意図がちょっとわからないので質問させてください。
③で取得したトークンを登録するだけだとどういう不都合が生じるのでしょうか？

https://ws-4020.github.io/mobile-app-crib-notes/react-native/santoku/application-architecture/push-notification/register-token

※ 内部で受けた質問の転載です

**Comment by kozake** [comment, marked as answer]
「登録トークンは定期的に更新する」というのが登録トークン管理のベストプラクティス としてFCMにて推奨されています。
その実現方法として次の方式としております。

・サーバでは登録から1ヶ月以上たっているデバイス登録トークンを返さない
・モバイルアプリではgetTokenで取得したものがなかったら再生成する

この２つの組み合わせで登録トークンの定期的な更新を実現しています。
登録トークンはアプリの初回起動時にFCM SDKによって生成され、以後変わることはありません。
そのため、getTokenの結果をそのまま使ってしまうと、「登録から1ヶ月以上たっているけど過去に登録してあったトークン」がそのまま送信されて、登録トークンが更新されないので再生成しております。
